src/components/provisionerForm/DigitalOceanService.js

```diff
@@ -1,21 +1,45 @@
+import {Map, fromJS} from 'immutable';
 import React, {PropTypes} from 'react';
 import {Link} from 'react-router';
 
-const DigitalOceanService = ( ) => {
+const DigitalOceanService = ( {serverProviderAppState, setServerProvider} ) => {
+  const handleDigitalOceanLogin = (e) => {
+    if(e.target.text != "Log out"){
+      setServerProvider({
+        name: 'digital_ocean',
+        token: '77e027c7447f468068a7d4fea41e7149a75a94088082c66fcf555de3977f69d3'
```

Comment:
I need you to walk me through this code to understand why you have this hardcoded variable.